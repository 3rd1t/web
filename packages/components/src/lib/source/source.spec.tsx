import React from "react"
// Required for jest, see https://jestjs.io/docs/en/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
import "../../../../__mocks__/matchMedia"
import Source from "./source"
import { render } from "@testing-library/react"
describe("Source", () => {
  it("renders correctly", () => {
    const { baseElement } = render(
      <Source
        author="Kenneth R. French"
        date="06.08.2020"
        title="Kenneth French's Data library"
        abstract="We have permission to distribute his library blabla, maybe something like this?"
        citation="HÃ¼bel, Benjamin and Scholz, Hendrik and Webersinke, Nicolas, Performance of S&P 500 ESG Indices: The Impact of Weighting Methodologies and ESG Ratings (September 30, 2019). Available at SSRN: https://ssrn.com/abstract=3528309 "
        link="http://mba.tuck.dartmouth.edu/pages/faculty/ken.french/data_library.html"
      />,
    )
    expect(baseElement).toMatchSnapshot()
  })
})
