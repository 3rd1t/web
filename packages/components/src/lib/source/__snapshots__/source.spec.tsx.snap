// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Source renders correctly 1`] = `
<body>
  <div>
    <div>
      <div
        class="flex flex-wrap py-8 md:flex-no-wrap"
      >
        <div
          class="flex flex-col flex-shrink-0 pr-2 mb-6 md:w-64 md:mb-0"
        >
          <span
            class="font-medium tracking-widest text-gray-900"
          >
            Kenneth R. French
          </span>
          <span
            class="mt-1 text-sm text-gray-600"
          >
            06.08.2020
          </span>
        </div>
        <div
          class="space-y-4 md:flex-grow"
        >
          <h2
            class="text-2xl text-gray-900"
          >
            Kenneth French's Data library
          </h2>
          <p
            class="text-gray-700"
          >
            We have permission to distribute his library blabla, maybe something like this?
          </p>
          <div
            class="flex items-center space-x-4"
          >
            <a
              class="text-xs text-gray-900 focus:outline-none hover:text-research-600"
              href="http://mba.tuck.dartmouth.edu/pages/faculty/ken.french/data_library.html"
              target="blank"
            >
              Go to source
            </a>
            <button
              class="text-xs text-gray-900 focus:outline-none hover:text-research-600"
            >
              Citation
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
